pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -0,0 +1,67 @@
+package inferredspan
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/serverless/tags"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	// tagInferredSpanTagSource is the key to the meta tag
+	// that lets us know whether this span should inherit its tags.
+	// Expected options are "lambda" and "self"
+	tagInferredSpanTagSource = "_inferred_span.tag_source"
+
+	// additional function specific tag keys to ignore
+	functionVersionTagKey = "function_version"
+	coldStartTagKey       = "cold_start"
+)
+
+var functionTagsToIgnore = [...]string{
+	tags.FunctionARNKey,
+	tags.FunctionNameKey,
+	tags.ExecutedVersionKey,
+	tags.EnvKey,
+	tags.VersionKey,
+	tags.ServiceKey,
+	tags.RuntimeKey,
+	tags.MemorySizeKey,
+	tags.ArchitectureKey,
+	functionVersionTagKey,
+	coldStartTagKey,
+}
+
+// Check determines if a span belongs to a managed service or not
+// _inferred_span.tag_source = "self" => managed service span
+// _inferred_span.tag_source = "lambda" or missing => lambda related span
+func Check(span *pb.Span) bool {
```

Comment:
nit : the body of this function can be replaced with `return strings.Compare(span.Meta[tagInferredSpanTagSource], "self") == 0`